From 358824641bca3341774645d80252ada44a8e2b16 Mon Sep 17 00:00:00 2001
From: Sam Mortimer <sam@mortimer.me.uk>
Date: Sat, 26 May 2018 18:02:55 -0700
Subject: [PATCH 6/8] Add led capability
 LIGHTS_ADJUSTABLE_BATTERY_LED_BRIGHTNESS

*) Many existing liblights support notification brightness
   control via the alpha channel but do not support similar
   for the battery led.

*) https://github.com/LineageOS/android_lineage-sdk/commit/8f7a4559ab4a81855f399eae32ebe4a3f531e8b2
   introduced a dependency on light capability
   LIGHTS_ADJUSTABLE_BATTERY_LED_BRIGHTNESS such that, when
   set, it was assumed that the battery led brightness is
   also liblights adjustable.  It turned out that this
   was not the case.

*) Create a new capability to allow devices to distinguish
   whether their liblights supports adjustable brightness
   control for notification and battery leds independently
   of one another.

Change-Id: Id95905f07128c78c6acacf3b60d71bb37f70bcfb
---
 lineage/res/res/values/config.xml                              | 1 +
 .../lineageos/internal/notification/LightsCapabilities.java    | 3 +++
 .../lineageos/internal/notification/LineageBatteryLights.java  | 2 +-
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/lineage/res/res/values/config.xml b/lineage/res/res/values/config.xml
index c525a65..acea046 100644
--- a/lineage/res/res/values/config.xml
+++ b/lineage/res/res/values/config.xml
@@ -67,6 +67,7 @@
          LIGHTS_SEGMENTED_BATTERY_LED = 16
          LIGHTS_ADJUSTABLE_NOTIFICATION_LED_BRIGHTNESS = 32
          LIGHTS_BATTERY_LED = 64
+         LIGHTS_ADJUSTABLE_BATTERY_LED_BRIGHTNESS = 128
 
          For example, a device with notification and battery lights that supports
          pulsating and RGB control would set this config to 75. -->
diff --git a/sdk/src/java/org/lineageos/internal/notification/LightsCapabilities.java b/sdk/src/java/org/lineageos/internal/notification/LightsCapabilities.java
index a1ee26a..04351f2 100644
--- a/sdk/src/java/org/lineageos/internal/notification/LightsCapabilities.java
+++ b/sdk/src/java/org/lineageos/internal/notification/LightsCapabilities.java
@@ -42,6 +42,9 @@ public final class LightsCapabilities {
     // Device has a battery light.
     public static final int LIGHTS_BATTERY_LED = 64;
 
+    // The battery light supports adjustable brightness.
+    public static final int LIGHTS_ADJUSTABLE_BATTERY_LED_BRIGHTNESS = 128;
+
     public static boolean supports(Context context, final int capability) {
         final int capabilities = context.getResources().getInteger(
                 org.lineageos.platform.internal.R.integer.config_deviceLightCapabilities);
diff --git a/sdk/src/java/org/lineageos/internal/notification/LineageBatteryLights.java b/sdk/src/java/org/lineageos/internal/notification/LineageBatteryLights.java
index a318ce1..121ac8e 100644
--- a/sdk/src/java/org/lineageos/internal/notification/LineageBatteryLights.java
+++ b/sdk/src/java/org/lineageos/internal/notification/LineageBatteryLights.java
@@ -81,7 +81,7 @@ public final class LineageBatteryLights {
                 mContext, LightsCapabilities.LIGHTS_RGB_BATTERY_LED);
 
         mHALAdjustableBrightness = LightsCapabilities.supports(
-                mContext, LightsCapabilities.LIGHTS_ADJUSTABLE_NOTIFICATION_LED_BRIGHTNESS);
+                mContext, LightsCapabilities.LIGHTS_ADJUSTABLE_BATTERY_LED_BRIGHTNESS);
 
         // We support brightness adjustment if either the HAL supports it
         // or the light is RGB adjustable.
-- 
2.17.0

